crates/evm/src/lib.rs

```diff
@@ -262,17 +279,65 @@ pub struct NextBlockEnvAttributes {
     pub gas_limit: u64,
 }
 
-/// Function hook that allows to modify a transaction environment.
-pub trait TxEnvOverrides {
-    /// Apply the overrides by modifying the given `TxEnv`.
-    fn apply(&mut self, env: &mut TxEnv);
+/// Abstraction over transaction environment.
+pub trait TransactionEnv:
+    Into<revm_primitives::TxEnv> + Debug + Default + Clone + Send + Sync + 'static
+{
+    /// Returns configured gas limit.
+    fn gas_limit(&self) -> u64;
+
+    /// Set the gas limit.
+    fn set_gas_limit(&mut self, gas_limit: u64);
+
+    /// Returns configured gas price.
+    fn gas_price(&self) -> U256;
+
+    /// Returns configured value.
+    fn value(&self) -> U256;
+
+    /// Caller of the transaction.
+    fn caller(&self) -> Address;
+
+    /// Set access list.
+    fn set_access_list(&mut self, access_list: AccessList);
```

Comment:
should we also have builder style functions so you can chain setters `wtih_access_list(mut self)->Self where Self: Sized`?